applications/MPMApplication/custom_utilities/material_point_generator_utility.cpp

```diff
@@ -517,7 +520,7 @@ namespace Kratos::MaterialPointGeneratorUtility
                                     // Mark as boundary condition
                                     p_condition->Set(BOUNDARY, true);
 
-                                    if (boundary_condition_type == 1)
+                                    if (boundary_condition_type == 1 || boundary_condition_type == 2 || boundary_condition_type == 3)
```

Comment:
Not sure if it makes sense, but maybe add a warning/error like in line 116 when the background grid is not quadrilateral/ hexahedra.
```suggestion
const GeometryData::KratosGeometryType background_geo_type = rBackgroundGridModelPart.ElementsBegin()->GetGeometry().GetGeometryType();
if (boundary_condition_type == 1 || boundary_condition_type == 2 || boundary_condition_type == 3)
{
    // throw error if it tries to generate Lagrange BC with triangular/tetrahedra
    if ((boundary_condition_type == 2 || boundary_condition_type == 3) && (background_geo_type == GeometryData::KratosGeometryType::Kratos_Quadrilateral2D4 || background_geo_type == GeometryData::KratosGeometryType::Kratos_Hexahedra3D8))
         {
         KRATOS_ERROR << "Lagrange multiplier condition is currently only suitable for quadrilateral/hexahedral elements. Boundary locking effect in case of triangular/tetrahedral elements"  << std::endl;
         }
    // otherwise do stuff
}
```

Reply:
I added an error in case of triangular/tetrahedral background grid elements